src/main/java/edu/hawaii/its/api/service/OotbGrouperApiService.java

```diff
@@ -75,6 +77,9 @@ public FindGroupsResults findGroupsResults(List<String> groupPaths) {
      */
     public SubjectsResults getSubjects(String uhIdentifier) {
         SubjectsResults subjectsResults = ootbGroupingPropertiesService.getSubjectsResults();
+
+        subjectsResults.getSubjectsAfterAssignSubject(uhIdentifier);
+
         return subjectsResults;
     }
```

Comment:
Is it possible to already create the `subjectResults` object in `ootbGroupingPropertiesService.getSubjectsResults()` without having to externalize it to a new method in `SubjectsResults`? For example, our unit tests use `JsonUtil` methods to transform test properties into the class we want.

Reply:
In ootbGroupingPropertiesService, it's possible not only to fetch subjectsResults directly from the data harness but also to modify it as I wish and even create objects on my own which mean I can do everything without having to externalize it to a new method in SubjectsResults . However, my intention isn't to replace the modified subjectsResults back into the bean since that would violate the singleton nature of the bean data. Instead, my aim is to make modifications directly on the object retrieved from the bean itself. My concern lies in potentially causing issues by repeatedly removing and reinserting the modified object into the bean. Therefore, to maintain the singleton pattern, I am considering writing the method in an internalized manner.

Reply:
If we need refactoring for better clean code structure, we can make separate ticket after implementing most of the method of Ootb service

Reply:
Sorry, I'm just brainstorming ideas. Maybe having OOTB wrapper classes like `OotbSubjectsResults` might be the way to go. If it is important to merge now and you want to make a separate ticket for refactoring, move back the code to OotbGrouperApiService. A principle I am pushing for is to keep OOTB code mostly out of the main code.

Reply:
Everything is fine. I also like brainstorming ideas. Making Ootb wrapper class was also one of the options that I talked with Frank before I started to implement ootb service. It didn't take much time to move the methods to wrapper class which means it would be not harder than I think to refactor the code later if we need. Thank you for good idea! I'm also trying to keep OOTB code out of the main code and my goal is kinda to make it easy for others to understand and access when more features are added later and need to be implemented in OOTB."